hash.go

```diff
@@ -0,0 +1,100 @@
+package bn256
+
+// HashG1 implements a hashing function into the G1 group. It uses Fouque-Tibouchi encoding described here
+// https://tools.ietf.org/pdf/draft-irtf-cfrg-hash-to-curve-04.pdf
+// Non-secret salt, optional (can be nil); recommended: hash-length random value.
```

Comment:
This might be renamed to `salt` -> `DST`, to recall it is the domain separation tag (DST).
the recommendation for generating a DST is on draft. Usually DST is the name of the protocol.

Reply:
done.